apps/passport/app/components/connect-button/ConnectButton.tsx

```
@@ -1,38 +1,89 @@
 import { Button } from '@kubelt/design-system/src/atoms/buttons/Button'
 import type { ButtonProps } from '@kubelt/design-system/src/atoms/buttons/Button'
 
 import walletsSvg from './wallets.png'
 import { Avatar } from 'connectkit'
+import { Spinner } from '@kubelt/design-system/src/atoms/spinner/Spinner'
+import { useAccount, useDisconnect, useSignMessage } from 'wagmi'
 
-const ConnectKitProvider = lazy(() =>
-  import('connectkit').then((module) => ({
-    default: module.ConnectKitProvider,
-  }))
-)
+import { ConnectKitProvider, ConnectKitButton } from 'connectkit'
 
-const CustomConnectKitButton = lazy(() =>
-  import('connectkit').then((module) => ({
-    default: module.ConnectKitButton.Custom,
-  }))
-)
+// const ConnectKitProvider = lazy(() =>
+//   import('connectkit').then((module) => ({
+//     default: module.ConnectKitProvider,
+//   }))
+// )
+
+// const CustomConnectKitButton = lazy(() =>
+//   import('connectkit').then((module) => ({
+//     default: module.ConnectKitButton.Custom,
+//   }))
+// )
 
 export type ConnectButtonProps = {
   connectCallback: (address: string) => void
+  signCallback: (
+    address: string,
+    signature: string,
+    nonce: string,
+    state: string
+  ) => void
   connectErrorCallback: (error: Error) => void
   disabled?: boolean
-  provider?: string
+  signData?: {
+    nonce: string
+    state: string
+    address: string
+  }
+  signing?: boolean
+  isLoading?: boolean
   className?: string
 } & ButtonProps
 
 export function ConnectButton({
   connectCallback,
+  connectErrorCallback,
+  signCallback,
+  isLoading,
   className,
+  signData,
 }: ConnectButtonProps) {
+  const { connector, isConnected, isReconnecting } = useAccount()
+  const { disconnect } = useDisconnect()
+  const {
+    isLoading: isSigning,
+    error,
+    signMessage,
+  } = useSignMessage({
+    onSuccess(data, variables) {
+      signCallback(signData?.address!, data, signData?.nonce!, signData?.state!)
```

Should throw an error just in one of the properties is missing?